File: sample-java/src/main/java/fr/bmartel/bboxapi/router/javasample/RemoteAccess.java
package fr.bmartel.bboxapi.router.javasample;

import com.github.kittinunf.fuel.core.FuelError;
import com.github.kittinunf.fuel.core.Request;
import com.github.kittinunf.fuel.core.Response;
import com.github.kittinunf.result.Result;
import fr.bmartel.bboxapi.router.BboxApiRouter;
import kotlin.Triple;

public class RemoteAccess {

    public static void main(String args[]) throws InterruptedException {
        BboxApiRouter bboxapi = new BboxApiRouter();
        bboxapi.setPassword("AAaaa*1111");
        Triple<Request, Response, Result<String, FuelError>> data = bboxapi.configureRemoteAccess(true);
        if (data != null) {
            Request request = data.getFirst();
            Response response = data.getSecond();
            Result<String, FuelError> obj = data.getThird();
            System.out.println(response.getStatusCode());
        } else {
            System.out.println("remote is not activable, change password strength to STRONG");
        }
    }
}

File: sample/src/main/kotlin/fr/bmartel/bboxapi/router/sample/RemoteAccess.kt
package fr.bmartel.bboxapi.router.sample

import com.github.kittinunf.fuel.core.FuelError
import com.github.kittinunf.result.Result
import fr.bmartel.bboxapi.router.BboxApiRouter

fun main(args: Array<String>) {
    val bboxapi = BboxApiRouter()
    bboxapi.password = "AAaaa*1111"
    val triple = bboxapi.configureRemoteAccess(state = false)
    if (triple != null) {
        when (triple.third) {
            is Result.Failure -> {
                val ex = (triple.third as Result.Failure<String, FuelError>).getException()
                println(ex)
            }
            is Result.Success -> {
                println(triple.second.statusCode)
            }
        }
    } else {
        println("remote is not activable, change password strength to STRONG")
    }
}